// Copyright 2019-present Open Networking Foundation
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto3";

package gRPCCellConfigReport;

service gRPCCellConfigReportUpdater {
    rpc UpdateCellConfig (gMsgCellConfigValues) returns (gMsgResults) {}
}

message gMsgECGI {
    string plmnId = 1;
    string ecid = 2;
}

message gMsgCandScell {
    string pci = 1;
    string earfcnDl = 2;
}

message gMsgCellConfigValues {
    gMsgECGI ecgi = 1;
    string pci = 2;
    repeated gMsgCandScell candScells = 3;
    string earfcnDl = 4;
    string earfcnUl = 5;
    string rbsPerTtiDl = 6;
    string rbsPerTtiUl = 7;
    string numTxAntenna = 8;
    string duplexMode = 9;
    string maxNumConnectedUes = 10;
    string maxNumConnectedBearers = 11;
    string maxNumUesSchedPerTtiDl = 12;
    string maxNumUesSchedPerTtiUl = 13;
    string dlfsSchedEnable = 14;
}

message gMsgResults {
    int32 result = 1;
}
